src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1046 @@
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_USER_ID));
+                    tokenInfo.addParameter(OktaConstants.ACCESS_TOKEN_IDENTIFIER,
+                            responseJSON.get(OktaConstants.ACCESS_TOKEN_IDENTIFIER));
+
+                    return tokenInfo;
+                }
+            } else {
+                log.error(String.format("Invalid token %s", accessToken));
+                tokenInfo.setTokenValid(false);
+                tokenInfo.setErrorcode(APIConstants.KeyValidationStatus.API_AUTH_INVALID_CREDENTIALS);
+                return tokenInfo;
+            }
+        } catch (ParseException e) {
+            handleException(OktaConstants.ERROR_WHILE_PARSE_RESPONSE + e.getMessage(), e);
+        } catch (UnsupportedEncodingException e) {
+            handleException(OktaConstants.ERROR_ENCODING_METHOD_NOT_SUPPORTED + e.getMessage(), e);
+        } catch (ClientProtocolException e) {
+            handleException("HTTP request error has occurred while sending request to OAuth provider. " +
+                    e.getMessage(), e);
+        } catch (IOException e) {
+            handleException(OktaConstants.ERROR_OCCURRED_WHILE_READ_OR_CLOSE_BUFFER_READER + e.getMessage(), e);
+        } finally {
+            closeResources(reader, httpClient);
+        }
+
+        return null;
+    }
+
+    @Override
+    public KeyManagerConfiguration getKeyManagerConfiguration() throws APIManagementException {
+        return configuration;
+    }
+
+    @Override
+    public OAuthApplicationInfo buildFromJSON(String s) throws APIManagementException {
+        return null;
+    }
+
+    /**
+     * This method will be called when mapping existing OAuth Clients with Application in API Manager
+     *
+     * @param oAuthAppRequest Details of the OAuth Client to be mapped.
+     * @return {@code OAuthApplicationInfo} with the details of the mapped client.
+     * @throws APIManagementException This is the custom exception class for API management.
+     */
+    @Override
+    public OAuthApplicationInfo mapOAuthApplication(OAuthAppRequest oAuthAppRequest) throws APIManagementException {
+        return oAuthAppRequest.getOAuthApplicationInfo();
+    }
+
+    @Override
+    public boolean registerNewResource(API api, Map resourceAttributes) throws APIManagementException {
+        return true;
+    }
+
+    @Override
+    public Map getResourceByApiId(String apiId) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public boolean updateRegisteredResource(API api, Map resourceAttributes) throws APIManagementException {
+        return true;
+    }
+
+    @Override
+    public void deleteRegisteredResourceByAPIId(String apiID) throws APIManagementException {
+        //Not applicable
+    }
+
+    @Override
+    public void deleteMappedApplication(String s) throws APIManagementException {
+        //Not applicable
```

Comment:
If we have not implemented shouldn't we throw the expection, NotSupportedOperationException? We can simply write a sub class of APIManagementException and throw it

Reply:
When we delete the application(Not a mapped application) this method will be invoked by the framework. If we throw an exception, this will shown  to the UI as a pop up message. This will confuse the user.